src/devices/vdml_adi.cpp

```diff
@@ -159,4 +159,34 @@ double ADIPotentiometer::get_angle() const {
 	return ext_adi_potentiometer_get_angle(merge_adi_ports(temp_smart, _adi_port));
 }
 
+ADILed::ADILed(std::uint8_t adi_port) : ADIPort(adi_port) {
```

Comment:
For the adi, we might want to consider making the buffer something we initialize with a size, and having the buffer be a class member. This would reduce overhead and automate the buffer creation. Of course, we could always also have overloads (like you already have) for C-api arrays.